staging/src/k8s.io/apiserver/pkg/endpoints/request/context.go

```diff
@@ -94,3 +97,12 @@ func AuditEventFrom(ctx context.Context) *audit.Event {
 	ev, _ := ctx.Value(auditKey).(*audit.Event)
 	return ev
 }
+
+func WithAuditAnnotations(parent context.Context, annotations map[string]string) context.Context {
+	return WithValue(parent, auditAnnotationsKey, annotations)
+}
+
+func AuditAnnotationsFrom(ctx context.Context) (map[string]string, bool) {
```

Comment:
because these get sucked into the audit event at creation time, it means that anything appending to these after the audit filter runs are throwing values into a map that won't get used, right?

Reply:
Right.  I could try to do some hacks to unset this map after they are consumed.

Anyone wanting to set audit annotations after `WithAudit` is expected to use the audit event directly.

Reply:
Attempted to address in a fixup, PTAL.

Reply:
Now this looks unsafe to use in authenticators/authorizers since this will be called in TokenReview and SubjectAccessReview registry.

Reply:
For authentication - I structured the code so authentication returns an intent to set annotations via the response, but it makes no attempt to actually set them - that is the responsibility of the caller.  Thus the code is perfectly safe to use in TokenReview since the only calls to functions that would panic (if used incorrectly) are in `WithAuthentication`.

For authorization - it would just use the audit event directly.  SAR is unaffected by this change.